testing/combined-cs/src/main/java/org/projectnessie/nessie/combined/CombinedNessieClientFactory.java

```diff
@@ -0,0 +1,186 @@
+          (builder, version) -> {
+            for (NessieClientCustomizer customizer : customizers) {
+              builder = customizer.configure(builder, version);
+            }
+            return builder;
+          };
+
+      return new ClientFactory(apiVersion, persist) {
+        @Override // Note: this object is not serializable
+        @Nonnull
+        @jakarta.annotation.Nonnull
+        public NessieApiV1 make(NessieClientCustomizer customizer) {
+          return super.make(
+              (builder, version) ->
+                  customizer.configure(testCustomizer.configure(builder, version), version));
+        }
+      };
+    }
+
+    // We use a serializable impl. here as a workaround for @QuarkusTest instances, whose parameters
+    // are deep-cloned by the Quarkus test extension.
+    return new ClientFactory(apiVersion, persist);
+  }
+
+  private Persist persist(ExtensionContext extensionContext) {
+    return extensionContext
+        .getStore(NAMESPACE)
+        .getOrComputeIfAbsent(
+            "persistSupplier",
+            x -> {
+              PersistFactory persistFactory = backend(extensionContext).createFactory();
+              return persistFactory.newPersist(StoreConfig.Adjustable.empty());
+            },
+            Persist.class);
+  }
+
+  private Backend backend(ExtensionContext extensionContext) {
+    return extensionContext
+        .getStore(NAMESPACE)
+        .getOrComputeIfAbsent(
+            "backendSupplier",
+            x -> {
+              BackendFactory<InmemoryBackendConfig> backendFactory = new InmemoryBackendFactory();
+              return backendFactory.buildBackend(backendFactory.newConfigInstance());
+            },
+            Backend.class);
+  }
+
+  @Override
+  public void beforeEach(ExtensionContext extensionContext) {
+    Logics.repositoryLogic(persist(extensionContext)).initialize("main");
+  }
+
+  static NessieApiVersion apiVersion(ExtensionContext context) {
+    return UniqueId.parse(context.getUniqueId()).getSegments().stream()
+        .filter(s -> API_VERSION_SEGMENT_TYPE.equals(s.getType()))
+        .map(UniqueId.Segment::getValue)
+        .findFirst()
+        .map(NessieApiVersion::valueOf)
+        .orElse(DEFAULT_API_VERSION);
+  }
+
+  private static class ClientFactory implements NessieClientFactory, Serializable {
+    private final NessieApiVersion apiVersion;
+    private final Persist persist;
+
+    private ClientFactory(NessieApiVersion apiVersion, Persist persist) {
+      this.apiVersion = apiVersion;
+      this.persist = persist;
+    }
+
+    @Override
+    public NessieApiVersion apiVersion() {
+      return apiVersion;
+    }
+
+    @Override
+    @Nonnull
+    @jakarta.annotation.Nonnull
+    public NessieApiV1 make(NessieClientCustomizer customizer) {
+      NessieClientBuilder clientBuilder =
+          createClientBuilder("Combined", null)
+              .asInstanceOf(CombinedClientBuilder.class)
```

Comment:
Could we somehow "combine" these calls... something like `createClientBuilder(Class<T> builder)`... the name looks redundant in this context (separate PR, of course, if you agree).